cal/cal/static/js/app.js

```diff
@@ -3,32 +3,120 @@ var analyticsApp = angular.module('analyticsApp', ['nvd3', 'ui.calendar']);
 analyticsApp.controller('LoggedInCtrl', function LoggedInController($scope) {
 });
 
-// Controller to generate graph data from chart directive, cumulative tags
 analyticsApp.controller('TagsCtrl', function($scope, $http){
   var url = '/v1/tags.json';
+  var tagUrl = '/v1/tags/';
+
   $scope.url = url;
   $scope.tags = [];
 
+  // Generate graph data
   $http({ method: 'GET', url: url }).
     success(function successCallback(data) {
-
       for (var i = 0; i < data.results.length; i++) {
         var tag = data.results[i];
         $scope.tags.push({
+          id: tag.id,
           label: tag.label,
           keywords: tag.keywords,
           hours: tag.hours
         });
       }
-
     });
 
+    this.addTag = function(tag) {
+      $http({
+        method: 'POST',
+        url: url,
+        data: $.param({
```

Comment:
I believe you can use `params: {...}` instead of mixing in jQuery with `data: $.param({`. See the arguments section in [here](https://docs.angularjs.org/api/ng/service/$http)